File: sol/775.py
class Solution:
    def isIdealPermutation(self, nums: List[int]) -> bool:
        ptr = 0
        minToCheck = 0
        while ptr < len(nums):
            if nums[ptr] == minToCheck:
                ptr += 1
                minToCheck += 1
            elif nums[ptr] == minToCheck + 1 and nums[ptr + 1] == minToCheck:
                ptr += 2
                minToCheck += 2
            else:
                return False
        return True

File: sol/775/775.cpp
struct Solution {
    bool isIdealPermutation(const vector<int>& nums) {
        int minToCheck = 0, thisNum;
        size_t ptr = 0, n = nums.size();
        while (ptr < n){
            thisNum = nums[ptr];
            if (thisNum == minToCheck){
                ++ptr;
                ++minToCheck;
            } else if (thisNum == minToCheck + 1 && nums[ptr + 1] == minToCheck){
                ++++ptr;
                ++++minToCheck;
            } else {
                return false;
            }
        }
        return true;
    }
};
